[BANKCON-4310] Add bottom sheet support to consent screen

carlosmuvi-stripe opened the pull request:
# Summary
<!-- Simple summary of what was changed. -->

# Motivation
<!-- Why are you making this change? If it's for fixing a bug, if possible, please include a code snippet or example project that demonstrates the issue. -->

# Testing
<!-- How was the code tested? Be as specific as possible. -->
- [ ] Added tests
- [ ] Modified tests
- [ ] Manually verified

# Screenshots
| Before  | After |
| ------------- | ------------- |
| *before screenshot*  | *after screenshot* |

# Changelog
<!-- Is this a notable change that affects users? If so, add a line to `CHANGELOG.md` and prefix the line with one of the following:
    - [Added] for new features.
    - [Changed] for changes in existing functionality.
    - [Deprecated] for soon-to-be removed features.
    - [Removed] for now removed features.
    - [Fixed] for any bug fixes.
    - [Security] in case of vulnerabilities.
-->

scottmathews-stripe reviewed (commented):
Just a couple rotation issues with the new bottomsheet

<img width="1017" alt="image" src="https://user-images.githubusercontent.com/107271832/176485068-fee0564d-ebeb-4adc-a7c5-236f64b90178.png">
<img width="1017" alt="image" src="https://user-images.githubusercontent.com/107271832/176485144-a0886bc4-673b-4200-b03f-88ebe77318a5.png">

carlosmuvi-stripe commented:
Both good catches again @scottmathews-stripe !

- Added `skipHalfExpanded = true` to avoid that intermediate step.
- Made content scrollable and fixed button at the bottom when no enough space available (see video)
https://user-images.githubusercontent.com/99293320/176561665-ecb86a3f-3065-4c50-975f-07ddd2951b35.mp4

scottmathews-stripe reviewed (commented):
One more thing I noticed - after opening the "data requested" bottom sheet, the back button will take you out of the entire activity. I expected it to only dismiss the bottom sheet.

carlosmuvi-stripe commented:
> One more thing I noticed - after opening the "data requested" bottom sheet, the back button will take you out of the entire activity. I expected it to only dismiss the bottom sheet.

Good catch again! Added a back handler to override back behavior when bottomsheet is open (Compose makes things easy!) https://github.com/stripe/stripe-android/pull/5229/commits/6b8bc9f559120e50c1e220c5ff48b5576e021d87
